### PR TITLE
Cart Component Code Refactor

### LINKED ISSUE #442: Code Refactor of Cart Component.
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe the feature

**Description**

I want to perform a code Refactor on the Cart.js file because the file contains lots of variables and codes that have never been used and also some parts are rewritten even though the reusable components are present.

I will remove all the unnecessary parts of codes from this file and try to follow the DRY (Don't Repeat Yourself) principle and use reusable components wherever they can possibly be implemented. 

### Add Screenshots

..

### Record Options

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSOC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by AlfiyaSiddique**
# Description
Refactor the code in the cart component by dividing Steps into different Component
Changes all the class attribute to className attribute in react js files

Fixes #442
closes #442 **Code Refactor of Cart Component.**

## Type of change
Code Refactor

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots

https://github.com/UBA-GCOEN/StichHub/assets/86224794/b9e9f3bd-a323-4ea6-86e8-80b013fb75a0

### CODE REVIEW

**Comment by UBA-GCOEN** [comment]
Hello @AlfiyaSiddique!
Thank you for submitting this pull request Your pull request will be reviewed by maintainers shortly. Meanwhile make sure you have referenced the respected issue in this pr Your pr will not be merged if it does not have reference to respected issue
please read our Contributing Guidelines https://github.com/UBA-GCOEN/StichHub/blob/main/CONTRIBUTING.md And Code of Conduct https://github.com/UBA-GCOEN/StichHub/blob/main/CODE_OF_CONDUCT.md

**Comment by AlfiyaSiddique** [comment]
Hello @Siddhant-Patil0203  

Just a request to review and merge this PR a bit earlier cause it contains many minor changes in different files, hence it might be difficult for me to resolve conflicts if occur. Hope you understand 🙂

**Comment by AlfiyaSiddique** [comment]
@Naresh-chandanbatve  OR @Siddhant-Patil0203 , Can you please review my PRs . I won't be able to work on my other issues cause they might cause huge merge conflicts when these PR get merged if I already made changes in them. So I'm waiting for these to get merged.

**Review by Naresh-chandanbatve — APPROVED** [review]

**Pull request merged**
